jsonrpc/src/schemas/trace.json

```
@@ -0,0 +1,89 @@
+{
+  "traceResultObject": {
+    "title": "traceResultObject",
+    "description": "Trace data for the result of tracing content request.",
+    "type": "object",
+    "required": [
+      "origin",
+      "targetId",
+      "receivedFrom",
+      "responses",
+      "metadata"
+    ],
+    "properties": {
+      "origin": {
+        "description": "Local Node ID",
+        "$ref": "#/components/schemas/uint256"
+      },
+      "targetId": {
+        "description": "Target content ID",
+        "$ref": "#/components/schemas/uint256"
+      },
+      "receivedFrom": {
+        "description": "Node ID from which the content was received.",
+        "$ref": "#/components/schemas/uint256"
+      },
+      "responses": {
+        "$ref": "#/components/schemas/traceResultResponses"
+      },
+      "metadata": {
+        "$ref": "#/components/schemas/traceResultMetadata"
+      },
+      "cancelled": {
+        "$ref": "#/components/schemas/listOfNodeIds"
+      }
+    }
+  },
+  "listOfNodeIds": {
+    "title": "listOfNodeIds",
+    "description": "Contains list of nodes IDs.",
+    "type": "array",
+    "items": {
+      "$ref": "#/components/schemas/uint256"
+    }
+  },
+  "traceResultResponses": {
+    "title": "traceResultResponses",
+    "description": "Contains a map from remote node ID, to the node IDs it responded with. For the Node ID that is also in the `receivedFrom` field, respondedWith must be empty array.",
```

```suggestion
    "description": "Contains a map of remote node IDs with the node IDs each node responded with. For the node ID that is in the `receivedFrom` field, `respondedWith` MUST be empty array.",
```

Attempt to make the sentence more clear. Not sure if I fully succeeded, heh.

Also, shouldn't the node that provided the content simply be ommited from this list instead of the empty array?

---

I don't think we should omit that node. Technically it is also a response. Just the one that didn't contain any new nodes, only the content we were looking for.